### PR TITLE
Public preview pnp merge

### PR DESCRIPTION
**Pull request by jspaith**
Merge pnp tree into public-preview.

### CODE REVIEW

**Comment by massand** [comment]
@jspaith not a blocker - it appears the e2e tests aren't enabled for the new work. It would be good to get a clean run with other preview bits.

**Comment by jspaith** [comment]
@massand - correct re e2e for PnP.  Basically by time they started becoming ready, the tree was locked down to minimize new changes.  All other e2e - other than sfc remove we've discussed - should be present in here.

There's still work item on old repo to bring in the e2e.  Assuming everything else looks good could you please sign off or let me know what's off?  Thanks

---
In reply to: [520516530](https://github.com/Azure/azure-iot-sdk-c/pull/1107#issuecomment-520516530) [](ancestors = 520516530)

File: digitaltwin_client/doc/building_sdk.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Building the Digital Twin Device SDK
+
+## Initial development environment setup
+To get setup to build applications that enable the Digital Twin for C:
+* Clone and perform initial cmake on the private GitHub repo containing the Digital Twin private preview SDK.
+
+
+```
```
Comment (jebrando) [inline digitaltwin_client/doc/building_sdk.md]: nitpick: add the code type at the end (Shell).

File: jenkins/ubuntu1604_c.sh [hunk 2]
```diff
@@ -12,6 +12,6 @@ build_root=$(cd "$(dirname "$0")/.." && pwd)
 cd $build_root
 
 # -- C --
-./build_all/linux/build.sh --run-unittests --run_valgrind --run-e2e-tests --run-sfc-tests --provisioning --use-edge-modules "$@" #-x
```
Comment (jebrando) [inline jenkins/ubuntu1604_c.sh]: >--run-sfc-tests [](start = 74, length = 15)

Why is this one removed?

Comment (jebrando) [reply jenkins/ubuntu1604_c.sh]: Here and all the files.

---
In reply to: [313150340](https://github.com/Azure/azure-iot-sdk-c/pull/1107#discussion_r313150340) [](ancestors = 313150340)

**Comment by jspaith** [comment]
>nitpick: add the code type at the end (Shell).

Not sure what you mean by this?

**Pull request merged**